tests/test_controls_view.py

```diff
@@ -0,0 +1,171 @@
+"""Test the Controls widget."""
+
+import pytest
+from PyQt6 import QtWidgets
+from RATapi.controls import Controls, fields
+
+from rascal2.widgets.controls import ControlsWidget, FitSettingsWidget
+
+
+class MockValidationError:
+    """A mock of a ValidationError list."""
+
+    def errors(self):
+        return [{"msg": "Validation error!"}]
+
+
+class MockModel:
+    """A mock Model."""
+
+    def __init__(self):
+        self.controls = Controls()
+
+
+class MockPresenter:
+    """A mock MainWindowPresenter class."""
+
+    def __init__(self):
+        self.model = MockModel()
+        self.controls = self.model.controls
+        self.allow_setting = True
+        self.terminal_interrupted = False
+
+    def getControlsAttribute(self, attr):
+        return getattr(self.controls, attr)
+
+    def editControls(self, setting, value):
+        if self.allow_setting:
+            setattr(self.controls, setting, value)
+            return True
+        self.last_validation_error = MockValidationError()
+        return False
+
+    def interrupt_terminal(self):
+        self.terminal_interrupted = True
+
+
+class MockWindowView(QtWidgets.QMainWindow):
+    """A mock MainWindowView class."""
+
+    def __init__(self):
+        super().__init__()
+        self.presenter = MockPresenter()
```

Comment:
I think this should be set to None and populated with a `unittest.mock.Mock` except if a mock is somehow not possible as is the case for the view. The `MockPresenter` class in my opinion has too much implementation for a unit test which is a code smell. I think the refactor suggested above should make it easier to do with a MagicMock